frontend/src/metabase/query_builder/components/filters/modals/BulkFilterSelect/BulkFilterSelect.tsx

```diff
@@ -79,9 +85,62 @@ const BulkFilterSelect = ({
   );
 };
 
-const getNewFilter = (query: StructuredQuery, dimension: Dimension) => {
+const getNewFilter = (query: StructuredQuery, dimension: Dimension): Filter => {
   const filter = new Filter([], null, dimension.query() ?? query);
   return filter.setDimension(dimension.mbql(), { useDefaultOperator: true });
 };
 
-export default BulkFilterSelect;
+export interface SegmentFilterSelectProps {
+  query: StructuredQuery;
+  filters?: Filter[];
+  segments: SegmentOption[];
+  onAddFilter: (filter: Filter) => void;
+  onRemoveFilter: (filter: Filter) => void;
+}
+
+export const SegmentFilterSelect = ({
+  query,
+  filters,
+  segments,
+  onAddFilter,
+  onRemoveFilter,
+}: SegmentFilterSelectProps): JSX.Element => {
+  const activeSegments = useMemo(() => {
+    return segments.filter(segment => {
+      return !!filters?.find(
+        filter => filter[0] === "segment" && filter[1] === segment.filter[1],
```

Comment:
nit: move this code to metabase-lib, perhaps `Filter`

Reply:
:+1: I think i can modify this to use the existing `structuredQuery.segments()`